server/src/main/java/org/elasticsearch/index/shard/IndexLongFieldRange.java

```diff
@@ -76,6 +76,13 @@ public boolean isComplete() {
         return shards == null;
     }
 
+    /**
+     * @return whether this range includes information from all shards and can be used meaningfully.
+     */
+    public boolean containsAllShardRanges() {
+        return isComplete() && this != IndexLongFieldRange.EMPTY && this != IndexLongFieldRange.UNKNOWN;
```

Comment:
I think EMPTY can be meaningfully handled by the can-match handler if there is a timestamp condition. Clearly an edge case, but seems wrong to not handle this?